packages/mui-material/src/Slider/Slider.d.ts

```diff
@@ -269,8 +269,10 @@ export interface SliderOwnProps {
 export interface SliderTypeMap<
   RootComponent extends React.ElementType = 'span',
   AdditionalProps = {},
+  Value extends number | number[] = number | number[],
+  OnChangeValue extends Value = Value,
 > {
-  props: AdditionalProps & SliderOwnProps;
+  props: AdditionalProps & SliderOwnProps<Value, OnChangeValue>;
```

Comment:
Why do we need both Value and OnChangeValue?

Reply:
I proposed it instead of the `NoInfer` type here: https://github.com/mui/material-ui/pull/38753#discussion_r1315914713. Got it from here: https://github.com/microsoft/TypeScript/issues/14829#issuecomment-288902999